stl/inc/__msvc_int128.hpp

```diff
@@ -1367,6 +1382,28 @@ struct _Signed128 : _Base128 {
         _Word[1] |= _That._Word[1];
         return *this;
     }
+
+    template <class _Ty, enable_if_t<is_floating_point_v<_Ty>, int> = 0>
+    _NODISCARD constexpr explicit operator _Ty() const noexcept {
+        bool _Negative = int64_t(_Word[1]) < 0;
+        const _Unsigned128 _Abs{_Word[0], (_Negative ? uint64_t(-int64_t(_Word[1])) : uint64_t(_Word[1]))};
+        auto _Result = static_cast<_Ty>(_Abs);
+        return _Negative ? -_Result : _Result;
+    }

```

Comment:
Please ensure that `static_assert(static_cast<double>(static_cast<std::_Signed128>(-1.0)) == -1.0);` passes.
```suggestion
    template <class _Ty, enable_if_t<is_floating_point_v<_Ty>, int> = 0>
    _NODISCARD constexpr explicit operator _Ty() const noexcept {
        const auto _Unsigned_self = static_cast<_Unsigned128>(*this);
        return static_cast<int64_t>(_Word[1]) < 0 ? -static_cast<_Ty>(-_Unsigned_self)
                                                  : static_cast<_Ty>(_Unsigned_self);
    }
```

Reply:
fixed in [`ec352e2` (#3559)](https://github.com/microsoft/STL/pull/3559/commits/ec352e2f1ee68ccb779998746731f1d7b37ca8f0)
thanks